src/structures/Guild.js

```diff
@@ -1010,6 +1030,62 @@ class Guild extends Base {
       .then(newData => this.client.actions.GuildUpdate.handle(newData).updated);
   }
 
+  /**
+   * Welcome channel data
+   * @typedef {Object} WelcomeChannelData
+   * @property {string} description The description to show for this welcome channel
+   * @property {GuildChannelResolvable} channel The channel to link for this welcome channel
```

Comment:
As for the type; it seems like you can only use text / news so, couldn't this be slightly stricter by only allowing `Snowflake|TextChannel|NewsChannel`? I suppose a `*Resolvable` could be created for this to make it look a bit nicer.

Reply:
had that thought, not too sure how to apply it. NewsOrTextChannelResolvable sounds rather weird, but I suppose that is what it'd be

Reply:
GuildTextChannelResolvable